packages/@aws-cdk/aws-lambda/lib/scalable-function-attribute.ts

```diff
@@ -0,0 +1,81 @@
+import * as appscaling from '@aws-cdk/aws-applicationautoscaling';
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Interface for scalable attributes
+ */
+export interface IScalableFunctionAttribute {
+  /**
+   * Scale out or in to keep utilization at a given level. The utilization is tracked by the
+   * LambdaProvisionedConcurrencyUtilization metric, emitted by lambda. See:
+   * https://docs.aws.amazon.com/lambda/latest/dg/monitoring-metrics.html#monitoring-metrics-concurrency
+   *
+   * Allowed values: 0.1 - 0.9.
+   */
+  scaleOnUtilization(props: UtilizationScalingOptions): void;
+  /**
+   * Scale out or in based on schedule.
+   */
+  scaleOnSchedule(id: string, actions: appscaling.ScalingSchedule): void;
+}
+
+/**
+ * Properties for enabling Lambda utilization tracking
+ */
+export interface UtilizationScalingOptions extends appscaling.BaseTargetTrackingProps {
+  /**
+   * Target utilization value for the attribute. For example, .5 indicates that 50 percent of allocated provisioned concurrency is in use.
+   *
+   * Allowed values: 0.1 - 0.9.
+   */
+  readonly targetUtilizationValue: number;
+}
+
+/**
+ * A scalable lambda alias attribute
+ */
+export class ScalableFunctionAttribute extends appscaling.BaseScalableAttribute {
+  constructor(scope: Construct, id: string, props: appscaling.BaseScalableAttributeProps) {
+    super(scope, id, props);
+  }
+
+  /**
+   * Scale out or in to keep utilization at a given level. The utilization is tracked by the
+   * LambdaProvisionedConcurrencyUtilization metric, emitted by lambda. See:
+   * https://docs.aws.amazon.com/lambda/latest/dg/monitoring-metrics.html#monitoring-metrics-concurrency
+   *
+   * Allowed values: 0.1 - 0.9.
+   */
+  public scaleOnUtilization(props: UtilizationScalingOptions) {
+    if (props.targetUtilizationValue < 0.1 || props.targetUtilizationValue > 0.9) {
+      throw new Error('TargetUtilizationValue should be between 0.1 and 0.9.');
+    }
+    super.doScaleToTrackMetric('Tracking', {
+      targetValue: props.targetUtilizationValue,
+      predefinedMetric: appscaling.PredefinedMetric.LAMBDA_PROVISIONED_CONCURRENCY_UTILIZATION,
+      ...props,
+    });
+  }
+
+  /**
+   * Scale out or in based on schedule.
+   */
+  public scaleOnSchedule(id: string, action: appscaling.ScalingSchedule) {
+    super.doScaleOnSchedule(id, action);
+  }
```

Comment:
Why is this needed?

Also, would it make sense to use the same naming convention as the base class for the other method (e.g. `doScaleOnUtilization`)

Reply:
`doScaleOnUtilization()` is a protected method so I need to make it public here. Also, the method body looks like this:

```ts
protected doScaleOnSchedule(id: string, props: ScalingSchedule) {
    this.target.scaleOnSchedule(id, props);
  }
```
I figure it makes more sense to call this method `scaleOnSchedule()`, and I'm not sure why the `protected` method renamed it to `do...`